File: C/Birthday_Cake_candles.c
#include <stdio.h>
int birthdayCakeCandles(int heights[],int n){
    int hc=0,c=0,i=n;   
    while(n--)
        if(c<heights[n])
            c=heights[n];
    while(i--)
        if(c==heights[i])
            ++hc;
    return hc;
}
int main(){
    int n;
    scanf("%d",&n);
    int candles[n],size=n;
    while(n--)
        scanf("%d",&candles[n]);
    printf("%d",birthdayCakeCandles(candles,size));
    return 0;
}

File: C#/Birthday Cake Candles.cs
public static int birthdayCakeCandles(List<int> candles)
{
    int count = 0;
    int max = 0;
    foreach (var candle in candles)
    {
        int num = candle;
        if(num > max)
        {
            max = num;
            count = 1;
        }
        else if(max == num)
        {
            count++;
        }
    }
    return count;
}